app/models/bookmark.rb

```diff
@@ -1,3 +1,6 @@
 class Bookmark < ActiveRecord::Base
   belongs_to :topic
+  has_many :likes, dependent: :destroy
+  has_one :user, through: :topic
```

Comment:
Review `BookmarkPolicy` to see if its methods are needed since adding the `has_one :user, through: :topic` association to `Bookmark`.